tests/behat/context/ApiContext.php

```diff
@@ -1635,6 +1650,271 @@ public function iShouldGetTheStoredJsonObject(string $name): void
+  /**
+   * @Then /^the response should contain projects in the following order:$/
+   */
+  public function responseShouldContainProjectsInTheFollowingOrder(TableNode $table): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+
+    $responseArray = json_decode($response->getContent(), true);
+    $returned_programs = $responseArray['projects'];
+    $expected_programs = $table->getHash();
+
+    Assert::assertEquals(count($returned_programs), count($expected_programs), 'Number of returned programs should be '.count($returned_programs));
+
+    for ($i = 0; $i < count($returned_programs); ++$i)
+    {
+      Assert::assertEquals($returned_programs[$i]['name'], $expected_programs[$i]['Name']);
+    }
+  }
+
+  /**
+   * @Then /^the response should have the projects model structure$/
+   */
+  public function responseShouldHaveProjectsModelStructure(): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+    $responseArray = json_decode($response->getContent(), true);
+
+    foreach ($this->programs_structure as $key)
+    {
+      Assert::assertArrayHasKey($key, $responseArray, 'Response should contain '.$key);
+    }
+
+    $returned_programs = $responseArray['projects'];
+
+    foreach ($returned_programs as $program)
+    {
+      Assert::assertEquals(count($program), count($this->program_structure),
+        'Number of program fields should be '.count($this->program_structure));
+      foreach ($this->program_structure as $key)
+      {
+        Assert::assertArrayHasKey($key, $program, 'Program should contain '.$key);
+        Assert::assertEquals($this->checkProjectFieldsValue($program, $key), true);
+      }
+    }
+  }
+
+  /**
+   * @Then /^The response should have the project model structure$/
+   */
+  public function responseShouldHaveProjectModelStructure(): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+    $returned_programs = json_decode($response->getContent(), true);
+
+    foreach ($returned_programs as $program)
+    {
+      Assert::assertEquals(count($program), count($this->program_structure),
+        'Number of program fields should be '.count($this->program_structure));
+      foreach ($this->program_structure as $key)
+      {
+        Assert::assertArrayHasKey($key, $program, 'Program should contain '.$key);
+        Assert::assertEquals($this->checkProjectFieldsValue($program, $key), true);
+      }
+    }
+  }
+
+  /**
+   * @Then /^the response should have the featured projects model structure$/
+   */
+  public function responseShouldHaveFeaturedProjectsModelStructure(): void
+  {
+    $response = $this->getKernelBrowser()->getResponse();
+    $responseArray = json_decode($response->getContent(), true);
+
+    foreach ($this->programs_structure as $key)
+    {
+      Assert::assertArrayHasKey($key, $responseArray, 'Response should contain '.$key);
+    }
+
+    $returned_programs = $responseArray['projects'];
+
+    foreach ($returned_programs as $program)
```

Comment:
might wanna rename program in this scenario to something like
featured_program_response. Since this object contains way less field than a Program $program object. 
Else it can be a little confusing at first.

also in the checkFeaturedProjectFieldsValue method and so on